opm/simulators/flow/KeywordValidation.cpp

```diff
@@ -0,0 +1,310 @@
+/*
+  Copyright 2016 Statoil ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <map>
+#include <opm/parser/eclipse/Parser/ErrorGuard.hpp>
+#include <opm/parser/eclipse/Parser/ParseContext.hpp>
+#include <opm/simulators/flow/KeywordValidation.hpp>
+
+namespace Opm
+{
+
+namespace KeywordValidation
+{
+
+    std::map<std::string, UnsupportedKeywordProperties> unsupported_keywords
```

Comment:
I realize this is still WIP - and there are of course many ways to go about this, but I would suggest creating a class:
```
class KeywordValidator {

...
private:
    std::map<std::string, UnsupportedKeywordProperties>unsupported_keywords;
};
```
And then build up that instance with unsupported keywords runtime (or have derived class `FlowKeywordValidator`which adds the default unsupported keywords in the constructor or ....). The advantages of this is that it will be simple to configure the validator runtime, it will be simple to have different validators for different simulators/situations and finally the validator can be tested that way.

Reply:
Indeed, after solidifying the approach a bit, it will become a class, in case we would like this to be reusable.